src/Theme.cpp

```diff
@@ -0,0 +1,411 @@
+    Color("TitleBgActive", ImGuiCol_TitleBgActive),
+    Color("TitleBgCollapsed", ImGuiCol_TitleBgCollapsed),
+    Color("MenuBarBg", ImGuiCol_MenuBarBg),
+    Color("ScrollbarBg", ImGuiCol_ScrollbarBg),
+    Color("ScrollbarGrab", ImGuiCol_ScrollbarGrab),
+    Color("ScrollbarGrabHovered", ImGuiCol_ScrollbarGrabHovered),
+    Color("ScrollbarGrabActive", ImGuiCol_ScrollbarGrabActive),
+    Color("CheckMark", ImGuiCol_CheckMark),
+    Color("SliderGrab", ImGuiCol_SliderGrab),
+    Color("SliderGrabActive", ImGuiCol_SliderGrabActive),
+    Color("Button", ImGuiCol_Button),
+    Color("ButtonHovered", ImGuiCol_ButtonHovered),
+    Color("ButtonActive", ImGuiCol_ButtonActive),
+    Color("Header", ImGuiCol_Header),
+    Color("HeaderHovered", ImGuiCol_HeaderHovered),
+    Color("HeaderActive", ImGuiCol_HeaderActive),
+    Color("Separator", ImGuiCol_Separator),
+    Color("SeparatorHovered", ImGuiCol_SeparatorHovered),
+    Color("SeparatorActive", ImGuiCol_SeparatorActive),
+    Color("ResizeGrip", ImGuiCol_ResizeGrip),
+    Color("ResizeGripHovered", ImGuiCol_ResizeGripHovered),
+    Color("ResizeGripActive", ImGuiCol_ResizeGripActive),
+    Color("Tab", ImGuiCol_Tab),
+    Color("TabHovered", ImGuiCol_TabHovered),
+    Color("TabActive", ImGuiCol_TabActive),
+    Color("TabUnfocused", ImGuiCol_TabUnfocused),
+    Color("TabUnfocusedActive", ImGuiCol_TabUnfocusedActive),
+    Color("DockingPreview", ImGuiCol_DockingPreview),
+    Color("DockingEmptyBg", ImGuiCol_DockingEmptyBg),
+    Color("PlotLines", ImGuiCol_PlotLines),
+    Color("PlotLinesHovered", ImGuiCol_PlotLinesHovered),
+    Color("PlotHistogram", ImGuiCol_PlotHistogram),
+    Color("PlotHistogramHovered", ImGuiCol_PlotHistogramHovered),
+    Color("TableHeaderBg", ImGuiCol_TableHeaderBg),
+    Color("TableBorderStrong", ImGuiCol_TableBorderStrong),
+    Color("TableBorderLight", ImGuiCol_TableBorderLight),
+    Color("TableRowBg", ImGuiCol_TableRowBg),
+    Color("TableRowBgAlt", ImGuiCol_TableRowBgAlt),
+    Color("TextSelectedBg", ImGuiCol_TextSelectedBg),
+    Color("DragDropTarget", ImGuiCol_DragDropTarget),
+    Color("NavHighlight", ImGuiCol_NavHighlight),
+    Color("NavWindowingHighlight", ImGuiCol_NavWindowingHighlight),
+    Color("NavWindowingDimBg", ImGuiCol_NavWindowingDimBg),
+    Color("ModalWindowDimBg", ImGuiCol_ModalWindowDimBg),
+};
+
+// Internal Load/Dump functions
+
+static const std::string SchemaVersionLabel = "ImGuiThemeVersion";
+static const std::string SchemaVersionValue = "1.0.0";
+
+static const std::string ThemeNameLabel = "ThemeName";
+static const std::string StyleSectionLabel = "style";
+static const std::string ColorsSectionLabel = "colors";
+
+static bool ThemeJsonFromStyle(json& aOutJson, const ImGuiStyle& aStyle)
+{
+    aOutJson = json{{SchemaVersionLabel, SchemaVersionValue}, {ThemeNameLabel, "<final merged theme in effect>"}, {StyleSectionLabel, {}}, {ColorsSectionLabel, {}}};
+
+    json& styleJson = aOutJson[StyleSectionLabel];
```

Comment:
prefer auto, const auto, auto&,... instead of explicit types everywhere where the type is deducible please.